great_expectations/experimental/datasources/sqlite_datasource.py

```diff
@@ -27,19 +27,37 @@
     BatchSortersDefinition,
     ColumnSplitter,
     DatetimeRange,
+    QueryAsset,
+    SQLAsset,
     SQLDatasource,
     SQLDatasourceError,
     TableAsset,
     _query_for_year_and_month,
 )
 
 
+def _add_year_and_month_splitter(
```

Comment:
Another strategy I tried was defining a mixin class that only defined the `add_year_and_month_splitter` that could be mixed in with the `SqliteTableAsset` and `SqliteQueryAsset`. This let me sidestep defining `add_year_and_month_splitter` on these classes. I ran into issues the with type of `self` unless I made the mixin class a subclass of `SQLAsset` but it felt strange mixing that into `SqliteQueryAsset` which is derived from `QueryAsset`.  It also made the return type a generic `SQLAsset` since the mixin wouldn't know about `[Table|Query]Asset`.

The main issue is the inheritance tree looks like:

```
SQLAsset -> TableAsset -> SqliteTableAsset
                 |
                 |-> QueryAsset -> SqliteQueryAsset
```

So the `Sqlite*Asset`s earliest common ancestor is `SQLAsset`. The mixin would let me spread the functionality across the hierarchy without worrying about the structure of this inheritance tree with the drawback of losing some type specificity.

I could play with that again if that is preferred. This way is a bit more verbose but is also more specific so I decided to go with that for now.

Reply:
It just occurred to me I could probably dynamically generate the `Sqlite*Asset` subclasses to reduce the boilerplate.
I don't want sqlite to block this PR though since the main value comes from the `SQLDatasource` changes so unless someone feels strongly or there are a lot of other changes required I may save this more a followup PR to prep for adding more splitters.

Reply:
@billdirks I would be interested in better understanding this explanation, even if the implementation is not changed.  One thought that came to me was that in order to avoid the same code that seemingly appears in the parent `SQLAsset` class for `add_year_and_month_splitter()` would be to call `super()` here and instantiate the output field by field with the present class (similarly to the way it is done in `DataAssistant` implementations for returning the corresponding `DataAssistantResult` type (something like this technique I thought. might help here).  Thanks.